esp/clients/wsdfuaccess/wsdfuaccess.cpp

```diff
@@ -522,20 +528,40 @@ StringBuffer &encodeDFUFileMeta(StringBuffer &metaInfoBlob, IPropertyTree *metaI
      * Should be part of the same configuration setup.
      */
 #ifdef _USE_OPENSSL
-    if (metaInfo->hasProp("keyPairName") && environment) // without it, meta data is not encrypted
+    if (metaInfo->hasProp("keyPairName") && (isContainerized() || environment)) // without it, meta data is not encrypted
     {
         MemoryBuffer metaInfoBlob;
         metaInfo->serialize(metaInfoBlob);
+        const char *keyPairName = metaInfo->queryProp("keyPairName"); // NB: in container mode, this is cert. name
```

Comment:
Picky, but I think maybe it's "in container mode, this is the  name of the secret containing the cert"?  Only care because the secret can contain additional items.

Reply:
correct. I'll change comment.